### PR TITLE
Add scikit-beam stack

### PR DESCRIPTION
**Pull request by ericdill**

### CODE REVIEW

File: recipes/lmfit/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  git_tag : {{ version }}
+
+build:
+  number: 0
+
+requirements:
+  build:
+    - python
+    - setuptools
+    - numpy
```
Comment (msarahan) [inline recipes/lmfit/meta.yaml]: does this use numpy's C api?  If so, you need to add x.x here and in the run req's to pin the version.

File: recipes/scikit-beam/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  git_rev: v{{ version }}
+
+build:
+  number: 0
+
+requirements:
+  build:
+    - python
+    - setuptools
+    - numpy
```
Comment (msarahan) [inline recipes/scikit-beam/meta.yaml]: same as above: does this need to be pinned?

File: recipes/xraylib/build.sh [hunk 3]
```diff
@@ -0,0 +1,5 @@
+autoreconf -i
+./configure --enable-python --disable-perl --disable-java \
+	    --disable-fortran2003 --disable-lua --prefix=$PREFIX
+make -j 4
```
Comment (msarahan) [inline recipes/xraylib/build.sh]: Please remove -j 4 - we don't know how many CPUs the CI build workers have.

File: recipes/xraylib/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  git_rev: xraylib-{{ version }}
+
+build:
+  number: 0
+
+requirements:
+  build:
+    - python
+    - setuptools
+    - numpy
```
Comment (msarahan) [inline recipes/xraylib/meta.yaml]: Same as above: do you need to pin numpy here?

File: recipes/xraylib/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    - scipy >=0.7.0
+
+test:
+  commands:
+    - xraylib
+  imports:
+    - xraylib
+
+about:
+  license: BSD
```
Comment (msarahan) [inline recipes/xraylib/meta.yaml]: 2-clause or 3?

File: recipes/scikit-beam/build.sh [hunk 6]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+
```
Comment (msarahan) [inline recipes/scikit-beam/build.sh]: kill this file and build.bat, and add

```
build:
    script: python setup.py install --single-version-externally-managed --record=record.txt
```

in meta.yaml

File: recipes/lmfit/meta.yaml (hunk 1 above)
Comment (ericdill) [inline recipes/lmfit/meta.yaml]: lmfit does not use numpy's C api. These are included in the build requirements because the setup.py for lmfit lists numpy and scipy in the install_requires.  As far as I understand, anything that is listed in the install_requires section of the setup.py needs to go in the build requirements or conda build will complain about setuptools downloading being disabled. Is that still true?

File: recipes/xraylib/build.sh (hunk 3 above)
Comment (ericdill) [inline recipes/xraylib/build.sh]: Right. Leaked in from locally working conda recipe. Good catch.

File: recipes/scikit-beam/build.sh (hunk 6 above)
Comment (ericdill) [inline recipes/scikit-beam/build.sh]: Will do. Is record.txt included in the conda package?

File: recipes/xraylib/meta.yaml (hunk 5 above)
Comment (ericdill) [inline recipes/xraylib/meta.yaml]: This is 3 clause, right? https://github.com/tschoonj/xraylib/blob/master/license_all.txt

File: recipes/xraylib/meta.yaml (hunk 4 above)
Comment (ericdill) [inline recipes/xraylib/meta.yaml]: I'm not sure. attn @tacaswell ?

File: recipes/xraylib/meta.yaml (hunk 5 above)
Comment (msarahan) [inline recipes/xraylib/meta.yaml]: Yep

File: recipes/xraylib/meta.yaml (hunk 4 above)
Comment (ericdill) [inline recipes/xraylib/meta.yaml]: @msarahan How would I know if this library is using the numpy C API?  There are numpy imports in the Cython files (import and cimport).

File: recipes/xraylib/meta.yaml (hunk 5 above)
Comment (ericdill) [inline recipes/xraylib/meta.yaml]: Ok thanks. Updating now.

File: recipes/scikit-beam/build.sh (hunk 6 above)
Comment (msarahan) [inline recipes/scikit-beam/build.sh]: I think so, but not completely certain.

File: recipes/lmfit/meta.yaml (hunk 1 above)
Comment (msarahan) [inline recipes/lmfit/meta.yaml]: Yes.

File: recipes/xraylib/meta.yaml (hunk 5 above)
Comment (ericdill) [inline recipes/xraylib/meta.yaml]: Does the order of "3-clause" and "BSD" matter? i.e., "3-clause BSD" vs "BSD 3-clause"

Comment (ocefpaf) [inline recipes/xraylib/meta.yaml]: Not really, but I like `BSD 3-Clause` (because that is how OSI spells it)

File: recipes/xraylib/meta.yaml (hunk 4 above)
Comment (msarahan) [inline recipes/xraylib/meta.yaml]: I think the fact that you have cimport numpy means you do.

File: recipes/xraylib/meta.yaml (hunk 5 above)
Comment (ericdill) [inline recipes/xraylib/meta.yaml]: Ok, thanks. I'll update

File: recipes/xraylib/meta.yaml (hunk 4 above)
Comment (ericdill) [inline recipes/xraylib/meta.yaml]: Ok according to the Cython section of the numpy 1.10.1 docs:

```
This module shows use of the cimport statement to load the definitions from the numpy.pxd header that ships with Cython. It looks like NumPy is imported twice; cimport only makes the NumPy C-API available, 
```

Given that, I think it should be assumed that this library *is* using the numpy C api and so 'numpy x.x' should be in the recipe

File: recipes/scikit-beam/meta.yaml (hunk 2 above)
Comment (ericdill) [inline recipes/scikit-beam/meta.yaml]: Looks like yes since we cimport numpy in scikit-beam

**Comment by ericdill** [comment]
I'm going to break out xraylib and build that separately since it is a little troublesome to get building.

File: recipes/scikit-beam/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+    url: https://github.com/scikit-beam/scikit-beam/archive/v{{ version }}.tar.gz
+    fn: scikit-beam-v{{ version }}.tar.gz
+    sha256: 084690482e418defbe36bb0969133d8de22d21b4a06e09c62ccb6b1717d2e2b4
+
+build:
+    number: 0
+    script: python setup.py build_ext -i && python setup.py install --single-version-externally-managed --record=record.txt
+
+requirements:
+    build:
```
Comment (jakirkham) [inline recipes/scikit-beam/meta.yaml]: Please add `toolchain`.

File: recipes/scikit-beam/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+    version: {{ version }}
+
+source:
+    url: https://github.com/scikit-beam/scikit-beam/archive/v{{ version }}.tar.gz
+    fn: scikit-beam-v{{ version }}.tar.gz
+    sha256: 084690482e418defbe36bb0969133d8de22d21b4a06e09c62ccb6b1717d2e2b4
+
+build:
+    number: 0
+    script: python setup.py build_ext -i && python setup.py install --single-version-externally-managed --record=record.txt
```
Comment (jakirkham) [inline recipes/scikit-beam/meta.yaml]: Please change to the following.

```
    script:
        # FIXME: This is a hack to make sure the environment is activated.
        # The reason this is required is due to the conda-build issue
        # mentioned below.
        #
        # https://github.com/conda/conda-build/issues/910
        #
        - source activate "${CONDA_DEFAULT_ENV}"  # [unix]
        - python setup.py build_ext -i
        - python setup.py install --single-version-externally-managed --record=record.txt
```

**Comment by ericdill** [comment]
Note: Needs lmfit 0.8.3. Testing a new build of 0.8.3 on the lmfit feedstock. Will restart this build once 0.8.3 is built and uploaded to anaconda.org/conda-forge

**Comment by ericdill** [comment]
Rebased and pushed to trigger CIs again, now that lmfit 0.8.3 is available.

**Comment by ericdill** [comment]
This recipe is skipping windows builds for now because v0.0.8 is not windows-compatible. Once v0.0.9 is tagged, the scikit-beam-feedstock can be updated to build windows binaries.  This is good to go.  Can I harass @ocefpaf or @jakirkham for a final review?

File: .travis.yml [hunk 9]
```diff
@@ -37,4 +37,3 @@ script:
         echo "Building all recipes.";
         source ./.CI/build_all;
       fi
-
```
Comment (ocefpaf) [inline .travis.yml]: I guess this is a remnant from a CI hack that you are no longer using, right?

File: recipes/scikit-beam/meta.yaml [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+build:
+  number: 0
+  script:
+    # FIXME: This is a hack to make sure the environment is activated.
+    # The reason this is required is due to the conda-build issue
+    # mentioned below.
+    #
+    # https://github.com/conda/conda-build/issues/910
+    #
+    - source activate "${CONDA_DEFAULT_ENV}"  # [unix]
```
Comment (ocefpaf) [inline recipes/scikit-beam/meta.yaml]: Is this still needed? https://github.com/conda/conda-build/issues/910 was closed.

File: recipes/scikit-beam/meta.yaml [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+        - skbeam.core.accumulators
+        - skbeam.core.constants
+        - skbeam.core.fitting
+        - skbeam.core.fitting.base
+        - skbeam.ext
+        # xraylib is not conda installable on windows yet  # [not win]
+        - skbeam.core.fitting    # [not win]
+        - skbeam.core.constants.xrf    # [not win]
+        # has linux native library imports  # [not win]
+        - skbeam.ext.ctrans    # [not win]
```
Comment (ocefpaf) [inline recipes/scikit-beam/meta.yaml]: I recommend you to add:

```yaml
    commands:
        - conda inspect linkages -n _test scikit-beam  # [not win]
        - conda inspect objects -n _test scikit-beam  # [osx]
```

in all packages that are linking to something. That helps to debug issues.

File: .travis.yml (hunk 9 above)
Comment (ericdill) [inline .travis.yml]: Indeed. checked out the current .travis.yml from master to undo this change

File: recipes/scikit-beam/meta.yaml (hunk 10 above)
Comment (ericdill) [inline recipes/scikit-beam/meta.yaml]: Nope, thanks. Removed that line and the FIXME script

File: recipes/scikit-beam/meta.yaml (hunk 11 above)
Comment (ericdill) [inline recipes/scikit-beam/meta.yaml]: That's a great suggestion. Added that in. Thanks!

**Comment by ericdill** [comment]
@ocefpaf thanks for the suggestions. Incorporated all of them into this PR

**Comment by ericdill** [comment]
Thanks @ocefpaf !

File: recipes/scikit-beam/meta.yaml [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+
+requirements:
+    build:
+        - toolchain
+        - python
+        - setuptools
+        - numpy x.x
+        - six
+        - cython
+        - autoconf    # [osx]
```
Comment (jakirkham) [inline recipes/scikit-beam/meta.yaml]: Why was this OS X only? Shouldn't it be used on Linux too?

Comment (ericdill) [inline recipes/scikit-beam/meta.yaml]: 🤷‍♂️ I'm not sure. Does `toolchain` include relevant things?
